### ISSUE TITLE
Rewards suddenly drop to 0 during training with A2C

### ISSUE THREAD

**Issue by ShaniGam** [ikostrikov/pytorch-a2c-ppo-acktr#159]
I've been using this repository for a while and it works great, but I noticed that in some cases after some time of training the rewards suddenly drop to 0 and go up again afterward. Does anyone have any idea how to solve this issue? (the learning rate is already very small so I don't think that decreasing it is the solution).

**Comment by ikostrikov** [comment]
For which environment does it happen?

**Comment by ShaniGam** [comment]
openai-retro, I added a new NES game called Road FIghter, I don't think it's available on their repository.

**Comment by ikostrikov** [comment]
Do you have a fork I can take a look at?

**Comment by ShaniGam** [comment]
No, but I will try to create one so you can reproduce the problem. I guess you haven't encountered this issue before.

**Comment by ikostrikov** [comment]
Could you also share your fork of this repository?

**Comment by ShaniGam** [comment]
Now I do:
[https://github.com/ShaniGam/pytorch-a2c-ppo-acktr](https://github.com/ShaniGam/pytorch-a2c-ppo-acktr)
There is actually another issue. I used to work with the previous version of your repository (from a few months ago) and it worked great for this game. The problem was the reproducibility of the results (even when setting the cudnn to false etc..) so I figured I should use this version instead. Unfortunately, this version is slower (even when I don't use `--cuda-deterministic`) and the results are not as good as they were in the previous version.
Do you have any idea how to make the previous version deterministic? It would help a lot! If not, would love some help improving this version to get better results.

**Comment by ikostrikov** [comment]
Is there a specific reason to use A2C? My guess that it breaks because of normalization.

A2C is extremely hard to tune. I highly recommend to remove normalization and use PPO instead.

**Comment by ShaniGam** [comment]
But it did work before, just wasn't deterministic.

**Comment by ikostrikov** [comment]
Can you point to the specific commit that is faster and works better? I will help me a lot to investigate why it become slower and what was changed specifically.

**Comment by ShaniGam** [comment]
I cloned it on June 29th.
Thank you so much for the help! I have a deadline coming up and I really need to solve this issue.

**Comment by ikostrikov** [comment]
Ok. I will compare the commits but I will take several days.

**Comment by ShaniGam** [comment]
Ok. If you somehow find why this version wasn't deterministic please tell me.

**Comment by ikostrikov** [comment]
Performance issues have been fixed here:
https://github.com/ikostrikov/pytorch-a2c-ppo-acktr/commit/bc3a68fd2127754f81a1cf3f93ffc81b08667ce3

I'm trying to find why did performance drop.

**Comment by ShaniGam** [comment]
Thanks! 
I switched to this version because I couldn't make the previous one deterministic but it seems like the determinism causes all the troubles, I can see now that the results are much better when I don't use the `--cuda-deterministic` flag.
